rclc/src/rclc/executor.c

```diff
@@ -673,20 +689,274 @@ static
 rclc_executor_handle_t *
 _rclc_executor_find_handle(
   rclc_executor_t * executor,
-  const void * rcl_handle)
+  rclc_executor_handle_type_t handle_type,
+  const void * rcl_handle,
+  void ** search_cache)
 {
-  for (rclc_executor_handle_t * test_handle = executor->handles;
-    test_handle < &executor->handles[executor->index];
-    test_handle++)
-  {
-    if (rcl_handle == rclc_executor_handle_get_ptr(test_handle)) {
-      return test_handle;
+  if (NULL == executor) {
```

Comment:
Good call, but that one needs to stay as a special-case; `_rclc_executor_find_handle` does not return a `rcl_ret_t`, so `RCL_CHECK_FOR_NULL_WITH_MSG` would cause a return type mismatch compile error.

For this function, `NULL` is a type-correct value with a valid meaning (no handle found), and every function that calls `static rclc_executor_handle_t * _rclc_executor_find_handle()` already uses `RCL_CHECK_FOR_NULL_WITH_MSG` to test the executor.